libs/blocks/global-navigation/utilities/menu/menu.js

```diff
@@ -173,6 +173,18 @@ const decoratePromo = (elem, index) => {
     elem.classList.add('feds-promo--dark');
   }
 
+  if (!isImageOnly) {
+    const imageElemMaybe = imageElem && elem.removeChild(elem.querySelector('.feds-promo-image'));
+    const innerContainer = document.createElement('div');
+    innerContainer.classList.add('feds-promo-inner-container');
+
+    innerContainer.append(...elem.children);
+
+    elem.appendChild(innerContainer);
+
+    if (imageElemMaybe) elem.appendChild(imageElemMaybe);
```

Comment:
Couldn't we do this a bit earlier on? Like when we define `const imageElem = elem.querySelector('picture');`, before starting to parse the links and image? We could remove the picture early on, wrap all the other content in a special class and then re-assemble everything under a wrapper or something like that. Again, I'd try to avoid manipulating content we've already decorated, if possible.